packages/runtime/src/next/draft-mode/index.tsx

```diff
@@ -0,0 +1,32 @@
+import { z } from 'zod'
+import { cookies, draftMode } from 'next/headers'
+export const MAKESWIFT_DRAFT_MODE_DATA_COOKIE = 'x-makeswift-draft-data'
+
+const makeswiftSiteVersionSchema = z.enum(['Live', 'Working'])
+export const MakeswiftSiteVersion = makeswiftSiteVersionSchema.Enum
+export type MakeswiftSiteVersion = z.infer<typeof makeswiftSiteVersionSchema>
+
+export const makeswiftDraftDataSchema = z.object({
+  makeswift: z.literal(true),
+  siteVersion: makeswiftSiteVersionSchema,
+})
```

Comment:
The draft data could just be the site version, by the way, since there is no chance of conflict with other Next.js stuff and also we control this cookie.